atspi-common/src/events/object.rs

```diff
@@ -1138,25 +1211,59 @@ impl From<StateChangedEvent> for EventBodyOwned {
 }
 
 impl_from_user_facing_event_for_interface_event_enum!(
-	ChildrenChangedEvent,
+	ChildrenInsertedEvent,
+	ObjectEvents,
+	ObjectEvents::ChildrenInserted
+);
+impl_from_user_facing_type_for_event_enum!(ChildrenInsertedEvent, Event::Object);
+impl_try_from_event_for_user_facing_type!(
+	ChildrenInsertedEvent,
+	ObjectEvents::ChildrenInserted,
+	Event::Object
+);
+event_test_cases!(ChildrenInsertedEvent);
+impl_to_dbus_message!(ChildrenInsertedEvent);
+impl_from_dbus_message!(ChildrenInsertedEvent);
+impl_event_properties!(ChildrenInsertedEvent);
+
+impl_from_user_facing_event_for_interface_event_enum!(
+	ChildrenDeletedEvent,
 	ObjectEvents,
-	ObjectEvents::ChildrenChanged
+	ObjectEvents::ChildrenDeleted
 );
-impl_from_user_facing_type_for_event_enum!(ChildrenChangedEvent, Event::Object);
+impl_from_user_facing_type_for_event_enum!(ChildrenDeletedEvent, Event::Object);
 impl_try_from_event_for_user_facing_type!(
-	ChildrenChangedEvent,
-	ObjectEvents::ChildrenChanged,
+	ChildrenDeletedEvent,
+	ObjectEvents::ChildrenDeleted,
 	Event::Object
 );
-event_test_cases!(ChildrenChangedEvent);
-impl_to_dbus_message!(ChildrenChangedEvent);
-impl_from_dbus_message!(ChildrenChangedEvent);
-impl_event_properties!(ChildrenChangedEvent);
-impl From<ChildrenChangedEvent> for EventBodyOwned {
-	fn from(event: ChildrenChangedEvent) -> Self {
+event_test_cases!(ChildrenDeletedEvent);
+impl_to_dbus_message!(ChildrenDeletedEvent);
+impl_from_dbus_message!(ChildrenDeletedEvent);
+impl_event_properties!(ChildrenDeletedEvent);
+
+impl From<ChildrenInsertedEvent> for EventBodyOwned {
+	fn from(event: ChildrenInsertedEvent) -> Self {
+		EventBodyOwned {
+			properties: std::collections::HashMap::new(),
+			kind: "insert".to_string(),
+			detail1: event.index_in_parent,
+			detail2: i32::default(),
+			// `OwnedValue` is constructed from the `ObjectRef`
+			// Only path to fail is to convert a Fd into an `OwnedValue`.
+			// Therefore, this is safe.
+			any_data: Value::from(event.child)
+				.try_into()
+				.expect("Failed to convert child to OwnedValue"),
+		}
+	}
+}
+
+impl From<ChildrenDeletedEvent> for EventBodyOwned {
+	fn from(event: ChildrenDeletedEvent) -> Self {
 		EventBodyOwned {
 			properties: std::collections::HashMap::new(),
-			kind: event.operation,
+			kind: "delete".to_string(),
```

Comment:
Should this not be "remove"?

I may be misunderstanding things.

Reply:
Okay, I just tried a bit of monitoring on the bus and found that "insert" and "delete" seem right for 'kind'.

The <major>:<minor> values only matter for the match-rule.. So we could construct unique match rules for these.

Reply:
You are right.